learn/inner_workings/datatypes.mdx

```diff
@@ -200,6 +200,68 @@ After flattening, it would look like this:
 
 Once all objects inside a document have been flattened, Meilisearch will continue processing it as described in the previous sections. For example, arrays will be flattened, and numeric and boolean values will be turned into strings.
 
+### Nested document querying and subdocuments
+
+Meilisearch has no concept of subdocuments and cannot perform nested document querying. In the previous example, the relationship between an appointment's date and doctor is lost when flattening the `appointments` array:
+
+```json
+…
+  "appointments.date": [
+    "2022-01-01",
+    "2019-01-01"
+  ],
+  "appointments.doctor": [
+    "Jester Lavorre",
+    "Dorian Storm"
+  ],
+…
+```
+
+This may lead to unexpected behavior during search. The following dataset shows two patients and their respective appointments:
+
+```json
+[
+  {
+    "id": 0,
+    "patient_name": "Imogen Temult",
+    "appointments": [
+      {
+        "date": "2022-01-01",
+        "doctor": "Jester Lavorre"
+      }
+    ]
+  },
+  {
+    "id": 1,
+    "patient_name": "Caleb Widowgast",
+    "appointments": [
+      {
+        "date": "2022-01-01",
+        "doctor": "Dorian Storm"
+      },
+      {
+        "date": "2023-01-01",
+        "doctor": "Jester Lavorre"
+      }
+    ]
+  }
+]
+```
+
+The following query returns patients `0` and `1`:
+
+```sh
+curl \
+  -X POST 'http://localhost:7700/indexes/movie_ratings/search' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "q": "",
+    "filter": "(appointments.date CONTAINS 2022-01-01 AND appointments.doctor CONTAINS 'Jester Lavorre')"
+  }'
+```
+
+Meilisearch is unable to only return patients who had an appointment with `Jester Lavorre` in `2022-01-01`. Instead, it returns patients who had an appointment with `Jester Lavorre`, and patients who had an appointment in `2022-01-01`.
```

Comment:
Could we add an example of a workaround? If a user can create a single value with both pieces of information, they will be able to filter by it:
```
{
    "id": 1,
    "patient_name": "Caleb Widowgast",
    "appointmentsMerged": [
      "Dorian Storm 2022-01-01",
      "Jester Lavorre 2023-01-01"
    ]
  }
```